connector/connect/src/main/scala/org/apache/spark/sql/connect/dsl/package.scala

```diff
@@ -175,6 +178,28 @@ package object dsl {
   }
 
   object plans { // scalastyle:ignore
+    implicit class DslMockRemoteSession(val session: MockRemoteSession) {
+      def range(
+          start: Option[Int],
+          end: Int,
+          step: Option[Int],
+          numPartitions: Option[Int]): Relation = {
+        val range = proto.Range.newBuilder()
```

Comment:
Note that I need to keep `proto.Range` as `Range` itself is a built-in scala class so we need `proto.` to differentiate for this special case.

Reply:
I've been explicitly requesting this a couple of times already, as a coding style to always prefix the proto generated classes with their `proto.` prefix. I know it uses a little bit more horizontal space, but at the same time it makes always clear where this particular element comes from which is tremendously useful because we're consistently using the different types between the catalyst API and Spark Connect in the same code paths.

Reply:
It makes sense for `SparkConnectPlanner` where Catalyst and Proto are both mixed together, and we are keeping the approach you are asking there.

However this is the Connect DSL that only deal with protos. No Catalyst included in this package: https://github.com/apache/spark/blob/9fc3aa0b1c092ab1f13b26582e3ece7440fbfc3b/connector/connect/src/main/scala/org/apache/spark/sql/connect/dsl/package.scala#L17

Reply:
As long as no catalyst is in this package this is good with me. Thanks for clarifying.